src/csharp/Microsoft.Spark/Utils/AssemblyLoader.cs

```diff
@@ -10,6 +10,46 @@
 
 namespace Microsoft.Spark.Utils
 {
+    internal static class AssemblySearchPathResolver
+    {
+        internal const string AssemblySearchPathsEnvVarName = "DOTNET_ASSEMBLY_SEARCH_PATHS";
+
+        // Returns the paths to search when loading assemblies in the following order of
+        // precedence:
+        // 1) Comma-separated paths specified in DOTNET_ASSEMBLY_SEARCH_PATHS environment variable.
+        //    - Note that if a path starts with ".", the working directory will be prepended.
+        // 2) The working directory.
+        // 3) The directory of the application.
+        internal static string[] GetAssemblySearchPaths()
+        {
+            var searchPaths = new List<string>();
+            string searchPathsStr =
+                Environment.GetEnvironmentVariable(AssemblySearchPathsEnvVarName);
+
+            if (!string.IsNullOrEmpty(searchPathsStr))
+            {
+                foreach (string searchPath in searchPathsStr.Split(','))
+                {
+                    string trimmedSearchPath = searchPath.Trim();
+                    if (trimmedSearchPath.StartsWith("."))
+                    {
+                        searchPaths.Add(
```

Comment:
Will this run into any issues on windows `C:\some\path\.\other\path` which is a combination of `Path.Combine("C:\some\path", ".\other\path")` ?  Same question for linux.

Reply:
I don't think it will, but asking just in case.

Reply:
This works on Linux, but I guess not on Windows?

Reply:
Sorry I meant I don't think it will run into any issues.

Reply:
Looks like this also works on Windows.